esphome/components/nextion/nextion_base.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+#include "esphome/core/defines.h"
+#include "esphome/core/color.h"
+#include "nextion_component_base.h"
+namespace esphome {
+namespace nextion {
+
+#define USE_TFT_UPLOAD
+
+//#define NEXTION_PROTOCOL_LOG  // For testing purposes
+
+#ifdef ESPHOME_LOG_HAS_VERY_VERBOSE
+#define NEXTION_PROTOCOL_LOG
+#endif
+
+#ifdef NEXTION_PROTOCOL_LOG
+#ifdef ESPHOME_LOG_HAS_VERY_VERBOSE
+#define ESP_LOGN(tag, ...) esph_log_vv(tag, __VA_ARGS__)
+#else
+#define ESP_LOGN(tag, ...) esph_log_d(tag, __VA_ARGS__)
+#endif
+#else
+#define ESP_LOGN(tag, ...) \
+  {}
+#endif
+
+class NextionBase;
+
+static const uint8_t LOOP_TIMEOUT_MS = 200;
+
+class NextionBase {
+ public:
+  virtual void add_no_result_to_queue_with_set(NextionComponentBase *component, int state_value) = 0;
+  virtual void add_no_result_to_queue_with_set(std::string variable_name, std::string variable_name_to_send,
```

Comment:
```suggestion
  virtual void add_no_result_to_queue_with_set(const std::string &variable_name, const std::string &variable_name_to_send,
```

nit: with the current code, the string buffer will be copied on each invocation. With const references, only the reference is passed and no copying takes places (applies to other methods here too)

Reply:
For some reason I thought C++ would always pass by reference on the backend since this is not a simple type. I can update these